src/Enable.Extensions.Queuing.Abstractions/IQueueClient.cs

```diff
@@ -130,6 +130,72 @@ public interface IQueueClient : IDisposable
+        /// <param name="sessionId">The session ID to attach to the message.</param>
+        /// <param name="cancellationToken">
+        /// A <see cref="CancellationToken"/> to observe while waiting for a task to complete.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// </returns>
+        Task EnqueueAsync(
+            byte[] content,
+            string sessionId,
+            CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Asynchronously enqueue a message on to the queue, with a session ID.
+        /// </summary>
+        /// <param name="content">The payload of the message to enqueue.</param>
+        /// <param name="sessionId">The session ID to attach to the message.</param>
+        /// <param name="cancellationToken">
+        /// A <see cref="CancellationToken"/> to observe while waiting for a task to complete.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// </returns>
+        Task EnqueueAsync(
+            string content,
+            string sessionId,
+            CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Asynchronously enqueue a message on to the queue, with a session ID.
+        /// </summary>
+        /// <typeparam name="T">The type of the payload to enqueue.</typeparam>
+        /// <param name="content">The payload of the message to enqueue.</param>
+        /// <param name="sessionId">The session ID to attach to the message.</param>
+        /// <param name="cancellationToken">
+        /// A <see cref="CancellationToken"/> to observe while waiting for a task to complete.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// </returns>
+        Task EnqueueAsync<T>(
+            T content,
+            string sessionId,
+            CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Asynchronously enqueue a batch of messages on to the queue, with a shared session ID.
+        /// </summary>
+        /// <typeparam name="T">The type of the payload to enqueue.</typeparam>
+        /// <param name="messages">The collection of payload items to enqueue.</param>
+        /// <param name="sessionId">The session ID to attach to the messages.</param>
+        /// <param name="cancellationToken">
+        /// A <see cref="CancellationToken"/> to observe while waiting for a task to complete.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// </returns>
+        Task EnqueueAsync<T>(
+            IEnumerable<T> messages,
+            string sessionId,
+            CancellationToken cancellationToken = default(CancellationToken));
+
```

Comment:
We've added a load of overloads of the `EnqueueAsync<T>` method here. I don't think this approach scales well - if we need to double the number of overloads each time we add support for a new message property. This is also not the way we added support for a message ID property.

I think that all the existing overloads except the one that takes an `IQueueMessage` are simple convenience methods that let us easily enqueue some common types without needing to worry about `IQueueMessages`. I think that if you want to go beyond the simplest use cases, needing to construct an `IQueueMessage` and setting, for example, a `SessionId` on this is probably as we need to support.

Can we therefore remove all these new overloads?